File: 11.py
class Solution(object):
    def maxArea(self, height):
       
        i,j=0,len(height)-1
        result=0
        while i<j:
            hmin=min(height[i],height[j])
            result=max(result,hmin*(j-i))
            if height[i]<height[j]:
                i+=1
            else:
                j-=1
        return result

File: 11.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0, j = height.size() - 1;
        int result = 0;

        while (i < j) {
            int width = j - i;
            int hmin = min(height[i], height[j]);
            result = max(result, width * hmin);

            
            if (height[i] < height[j]) {
                i++;
            } else {
                j--;
            }
        }

        return result;
    }
};
